controllers/dspipeline_params.go

```diff
@@ -145,6 +147,28 @@ func (p *DSPAParams) ObjectStorageHealthCheckDisabled(dsp *dspa.DataSciencePipel
 	return false
 }
 
+// ExternalRouteEnabled will return true if an external route is enabled in the CR, otherwise false.
+func (p *DSPAParams) ExternalRouteEnabled(dsp *dspa.DataSciencePipelinesApplication) bool {
+	if dsp.Spec.ObjectStorage != nil {
+		return dsp.Spec.ObjectStorage.EnableExternalRoute
+	}
+	return false
+}
+
+func (p *DSPAParams) RetrieveAndSetExternalRoute(ctx context.Context, client client.Client, log logr.Logger) (*routev1.Route, error) {
+	// Retrieve the external route
+	route := &routev1.Route{}
+	namespacedName := types.NamespacedName{
+		Name:      "minio-" + p.Name,
+		Namespace: p.Namespace,
+	}
+	if err := client.Get(ctx, namespacedName, route); err != nil {
+		log.Info("Unable to retrieve route", "error", err)
+	}
+
+	return route, nil
```

Comment:
We may have gone over this already, but now that we've removed the execution-stop should we return `err` here?  otherwise, this function never will return anything other than `nil` for its `error`, and the caller would have an unused check as well (L400)

Reply:
I agree with you, will update it.